build/pipelines/templates/run-wad-tests.yaml

```diff
@@ -0,0 +1,173 @@
+    BuildConfiguration: Release
+    
+jobs:
+
+- job: Job_1
+  displayName: Build Deployment
+  condition: succeeded()
+  pool:
+    name: Hosted VS2017
+    demands:
+    - msbuild
+    - visualstudio
+
+  steps:
+  - task: NuGetToolInstaller@0
+    displayName: 'Use NuGet 4.9.4'
+    inputs:
+      versionSpec: 4.9.4
+      checkLatest: true
+
+  - task: NuGetCommand@2
+    displayName: 'NuGet restore (src/nuget.config)'
+    inputs:
+      restoreSolution: src/Calculator.sln
+      feedsToUse: config
+      nugetConfigPath: src/nuget.config
+
+  - task: NuGetCommand@2
+    displayName: 'NuGet restore (nuget.config)'
+    inputs:
+      restoreSolution: src/Calculator.sln
+      feedsToUse: config
+      nugetConfigPath: nuget.config
+
+  - task: NuGetCommand@2
+    displayName: 'NuGet restore (WinAppDriver)'
+    inputs:
+      restoreSolution: src/CalculatorWADTests/CalculatorTest.sln
+      feedsToUse: config
+
+  - task: VSBuild@1
+    displayName: 'Build WinAppDriverTests'
+    inputs:
+      solution: src/CalculatorWADTests/CalculatorTest.sln
+      platform: Any CPU
+      configuration: '$(BuildConfiguration)'
+    continueOnError: true
+
+  - task: VSBuild@1
+    displayName: 'Build solution Calculator'
+    inputs:
+      solution: src/Calculator.sln
+      msbuildArgs: '/p:Configuration=Deploy /p:AppxPackageDir="$(Build.ArtifactStagingDirectory)\AppxPackages\\" /p:UapAppxPackageBuildMode=CI/p:AppxBundlePlatforms="x64" /p:AppxBundle=Always'
+      platform: x86
+      configuration: '$(BuildConfiguration)'
+
+  - task: CopyFiles@2
+    displayName: 'Copy Files to: $(build.artifactstagingdirectory)\AppxPackages'
+    inputs:
+      SourceFolder: '$(system.defaultworkingdirectory)'
+      Contents: '**\bin\$(BuildConfiguration)\**'
+      TargetFolder: '$(build.artifactstagingdirectory)\AppxPackages'
+
+  - task: PublishSymbols@2
+    displayName: 'Publish symbols path'
+    inputs:
+      SearchPattern: '**\bin\**\*.pdb'
+      PublishSymbols: false
+    continueOnError: true
+
+  - task: PublishBuildArtifacts@1
+    displayName: 'Publish Artifact: drop'
+    inputs:
+      PathtoPublish: '$(build.artifactstagingdirectory)\AppxPackages\'
+
+
+- job: Job_2
+  displayName: Test Automation
+  dependsOn: Job_1
+  condition: succeeded()
+  pool:
+    name: Hosted Windows 2019 with VS2019
+    demands: DotNetFramework
+
+  steps:
+  - task: PowerShell@1
+    displayName: 'Set Resolution'
+    inputs:
+      scriptType: inlineScript
+      arguments: '-NonInteractive -Force:$true  -Verb RunAs'
+      inlineScript: 'Set-DisplayResolution -Width 1920 -Height 1080 -Force'
+    continueOnError: true
+
+  - task: DownloadBuildArtifacts@0
+    displayName: 'Download Build Artifacts'
+    inputs:
+      artifactName: drop
+      downloadPath: '$(build.artifactstagingdirectory)\AppxPackages'
+
+  - task: DotNetCoreInstaller@0
+    displayName: 'Use .NET Core sdk 1.0.4'
+
+  - task: VisualStudioTestPlatformInstaller@1
```

Comment:
Unfortunately VS Test is **not** configured by default on the Hosted VS2019 agent, [thus it is recommended to have a VS Tools Installer Task for each build environment](https://github.com/Microsoft/azure-pipelines-tasks/issues/8665) (though it isn't re-downloaded every time, just configured apparently).

Reply:
The hosted agents have Visual Studio installed, and Visual Studio includes VSTest. To use the version that came with Visual Studio instead of the version that the tools installer downloaded, you'd set the `vsTestVersion` input to the `VSTest` task to `16.0` instead of `toolsInstaller`. Does that work?

The customer in [that issue](https://github.com/Microsoft/azure-pipelines-tasks/issues/8665) is talking about something different--it looks like they have an on-prem setup (not hosted vs2019) and they want VSTest to be installed in one release definition phase and used in a later phase.

Reply:
I believe you're completely right here! I configured the VSTest task to run on Visual Studio 2019 (it was configured to "Installed By Tools Installer" before and that seems to be working **just fine now** on the hosted VS2019 agent.  

Marking this as resolved and submitting a new PR.